### DISCUSSION TITLE
2021-08-17 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#574] [Maintainer discussions]
### Discussion topics

- **[Retrospective](https://github.com/centerofci/mathesar/discussions/574#discussioncomment-1196596)**: Summary of last week's work.
- **[Active Work](https://github.com/centerofci/mathesar/discussions/574#discussioncomment-1196601)**: Work planned for the next week.
- **[Updates](https://github.com/centerofci/mathesar/discussions/574#discussioncomment-1196631)**: General updates everyone should be aware of.
- **[Questions](https://github.com/centerofci/mathesar/discussions/574#discussioncomment-1196643)**: Questions for the team to answer.

### About
Each Tuesday, we start a discussion about Mathesar's progress. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

**Comment by kgodey** [comment]
## Retrospective

This post lists issue updates from 2021-08-10 until now by milestone.

### [03. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6)

- **Issues started, not yet finished:**
  - [@pavish] #518

### [04. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1)

- **Issues closed:**
  - [@kgodey] #565
  - [@kgodey] #546
  - [@pavish] #296
- **Issues started, not yet finished:**
  - [@pavish] #295
- **New issues created:**
  - #568, created by @kgodey
  - #565, created by @pavish
  - #562, created by @pavish
  - #546, created by @kgodey

### [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **Issues closed:**
  - [@pavish] #548
  - [@pavish] #524
- **Issues started, not yet finished:**
  - [@pavish] #195
- **New issues created:**
  - #548, created by @pavish

### [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29)

- **Issues closed:**
  - [@andyharley-cci] #534
- **Issues started, not yet finished:**
  - [@andyharley-cci] #421
  - [@mr-gabe49] #483
  - [@kgodey] #358
  - [@andyharley-cci] #225
  - [@Dhruvi16] #322
- **New issues created:**
  - #572, created by @eito-fis
  - #566, created by @pavish
  - #564, created by @pavish
  - #563, created by @pavish
  - #560, created by @pavish
  - #559, created by @pavish
  - #534, created by @andyharley-cci

### [06. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **Issues closed:**
  - [@eito-fis] #452
  - #391
- **Issues started, not yet finished:**
  - [@eito-fis] #501
  - [@eito-fis] #494
  - [@pegnott] #495
  - [@eito-fis] #515

### [07. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **Issues closed:**
  - [@kgodey] #557
  - [@mathemancer] #434
  - [@mathemancer] #409
  - [@mathemancer] #383
  - [@mathemancer] #382
  - [@mathemancer] #381
  - [@mathemancer] #380
  - [@mathemancer] #379
  - [@mathemancer] #378
  - [@kgodey] #369
  - [@kgodey] #368
  - [@kgodey] #367
  - [@kgodey] #269
- **Issues started, not yet finished:**
  - [@powellc] #387
  - [@mathemancer] #417
  - [@ghislaineguerin] #238
  - [@ghislaineguerin] #537
  - [@ghislaineguerin] #237
  - [@ghislaineguerin] #240
  - [@ghislaineguerin] #235
  - [@ghislaineguerin] #236
  - [@ghislaineguerin] #241
  - [@ghislaineguerin] #239
- **New issues created:**
  - #557, created by @kgodey
  - #552, created by @kgodey
  - #537, created by @kgodey

### [08. Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **Issues started, not yet finished:**
  - [@ghislaineguerin] #243

### [09. Data Modeling](https://github.com/centerofci/mathesar/milestone/24)

- **New issues created:**
  - #538, created by @kgodey

### [10. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **Issues started, not yet finished:**
  - [@aaditkamat] #395
- **New issues created:**
  - #558, created by @kgodey

### [2021-09 improvements](https://github.com/centerofci/mathesar/milestone/30)

- **New issues created:**
  - #570, created by @asharonbaltazar
  - #555, created by @eito-fis

**Comment by kgodey** [comment]
## Active Work

This post lists active work based on the [Active](https://github.com/centerofci/mathesar/projects/1) GitHub project. It also includes a current summary of milestones involved in active work.

### [2021-08-18] [03. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6)

- **[13 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2203.%20Working%20with%20Schemas%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish] #518

### [2021-08-20] [04. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1)

- **[30 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2204.%20Tables%20from%20File%20Import%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish] #295
- **Issues not yet started:**
  - [@kgodey] #562
  - [@pavish] #568

### [2021-08-25] [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **[8 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2205.%20Editable%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish] #195
- **Issues not yet started:**
  - [@pavish] #197

### [2021-08-31] [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29)

- **[6 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%222021-08%20improvements%22) already closed.** 
- **Issues currently in progress:**
  - [@andyharley-cci] #421
  - [@mr-gabe49] #483
  - [@kgodey] #358
  - [@andyharley-cci] #225
  - [@Dhruvi16] #322
- **Issues not yet started:**
  - [@kgodey] #564
  - [@kgodey] #563
  - [@kgodey] #566
  - [@arvind-1] #313
- **Issues remaining in backlog:**
  - 5 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%222021-08%20improvements%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%222021-08%20improvements%22) issues
  - 10 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%222021-08%20improvements%22) issues
  - 6 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%222021-08%20improvements%22) issues
  - 2 [infrastructure](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20infrastructure%22+milestone%3A%222021-08%20improvements%22) issues

### [2021-09-06] [06. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **[16 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2206.%20Working%20with%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@eito-fis] #501
  - [@eito-fis] #494
  - [@pegnott] #495
  - [@eito-fis] #515
- **Issues not yet started:**
  - [@asharonbaltazar] #498
- **Issues remaining in backlog:**
  - 13 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2206.%20Working%20with%20Tables%22) issues
  - 1 other issues

### [2021-09-13] [07. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **[17 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2207.%20Initial%20Data%20Types%22) already closed.** 
- **Issues currently in progress:**
  - [@powellc] #387
  - [@mathemancer] #417
  - [@ghislaineguerin] #238
  - [@ghislaineguerin] #537
  - [@ghislaineguerin] #237
  - [@ghislaineguerin] #240
  - [@ghislaineguerin] #235
  - [@ghislaineguerin] #236
  - [@ghislaineguerin] #241
  - [@ghislaineguerin] #239
- **Issues not yet started:**
  - [@powellc] #388
  - [@powellc] #247
  - [@eito-fis] #392
  - [@eito-fis] #424
  - [@eito-fis] #425
  - [@eito-fis] #426
  - [@eito-fis] #427
  - [@eito-fis] #429
  - [@rootxnet] #404
  - [@rootxnet] #405
  - [@rootxnet] #407
  - [@rootxnet] #406
  - [@rootxnet] #408
  - [@kgodey] #410
  - [@kgodey] #411
  - [@powellc] #384
  - [@powellc] #385
  - [@kgodey] #552
- **Issues remaining in backlog:**
  - 20 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2207.%20Initial%20Data%20Types%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2207.%20Initial%20Data%20Types%22) issues
  - 8 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2207.%20Initial%20Data%20Types%22) issues

### [2021-09-16] [08. Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2208.%20Working%20with%20Views%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin] #243
- **Issues not yet started:**
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #455
  - [@ghislaineguerin] #456
  - [@ghislaineguerin] #466
- **Issues remaining in backlog:**
  - 8 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2208.%20Working%20with%20Views%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2208.%20Working%20with%20Views%22) issues
  - 1 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2208.%20Working%20with%20Views%22) issues
  - 1 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2208.%20Working%20with%20Views%22) issues
  - 1 other issues

### [2021-09-20] [09. Data Modeling](https://github.com/centerofci/mathesar/milestone/24)

- **[3 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2209.%20Data%20Modeling%22) already closed.** 
- **Issues not yet started:**
  - [@kgodey] #538
- **Issues remaining in backlog:**
  - 4 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2209.%20Data%20Modeling%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2209.%20Data%20Modeling%22) issues
  - 3 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2209.%20Data%20Modeling%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2209.%20Data%20Modeling%22) issues

### [2021-09-22] [10. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **[7 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2210.%20Multiple%20Database%20Support%22) already closed.** 
- **Issues currently in progress:**
  - [@aaditkamat] #395
- **Issues remaining in backlog:**
  - 4 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) issues
  - 3 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) issues

**Comment by kgodey** [comment]
## Updates

### Team
- @mathemancer is on parental leave for a couple of weeks starting today.
- This is @eito-fis's last week as an intern, but he'll be staying on as a member of the core team and contributing when he has time.

### Roadmap
- Our [milestone deadlines](https://github.com/centerofci/mathesar/milestones?direction=asc&sort=due_date&state=open) have been adjusted again. We still have  [04. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1) and [03. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6) due this week.
- We're going to pause frontend feature work after milestone 5 (editing records) to focus on tests and stability, as part of our August improvements milestone. Issues for this will be created by @pavish soon.

**Comment by kgodey** [comment]
## Questions

@eito-fis, @ghislaineguerin, @pavish: Your responses are required. Others are also welcome to respond.

### Retrospective
- What went well this week?
- What could have gone better this week?
- Are there any improvements to our process you'd like to suggest?

### Active Work
- Does your workload for this week look okay?
- We have two milestones due in the next week. Are there any blockers to meeting our upcoming deadlines?

### Discussion
- Anything else you'd like to discuss?

**Reply by ghislaineguerin** [reply]
> What went well this week?

I'm not sure, overall things got done, but I can't make much progress due to the number of pending reviews I have open.

> What could have gone better this week?

I should have locked a day to only focus on closing all reviews. 

> Are there any improvements to our process you'd like to suggest?

Maybe I need to try a shorter period for review, from a week to a half week. 

> Does your workload for this week look okay?

Yes, it does.

> We have two milestones due in the next week. Are there any blockers to meeting our upcoming deadlines?

None that I can think of.

> Anything else you'd like to discuss?

Nothing comes to mind right now. Thanks

**Reply by kgodey** [reply]
> What went well this week?

I closed a bunch of issues, including the type mapping meta-issue which sets us up for consistent use of types in the future.

> What could have gone better this week? 

I didn't make any progress on speccing out new milestones. It's not blocking anything yet, but it will block design if I don't get to it this week.

> Are there any improvements to our process you'd like to suggest?

None.

> Does your workload for this week look okay?

Yes, but I'm not 100% sure I can get it all done.

> We have two milestones due in the next week. Are there any blockers to meeting our upcoming deadlines?

It depends on how work on #562 goes. It is not straightforward to implement because it involves a bunch of alter statements in a transaction. We will need to move out the deadline for [04. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1) if I can't get the PR done by tomorrow.

**Reply by pavish** [reply]
> What went well this week?

The layout/structural changes needed for pages to be schema centric is complete. The base changes for file import preview page and indicators was finally done. I also had to focus more on interviews, PR reviews, new issues, during the last week.

> What could have gone better this week?

I could not focus much on all the design discussions, since there were a lot of them open. It takes considerable time to go through the design discussions and think it over. I did not get time to balance this.

> Are there any improvements to our process you'd like to suggest?

This was discussed on Matrix. I will reiterate them here:
* I would like all new frontend issues to be updated with the apis needed for them. (I think it might be worth it to also update existing issues, since I had no idea of all the apis that were present. eg., `type_suggestions`).
* Before the start of each milestone, I would like to spec out frontend issues, go over the associated apis, and test them out. This will help us identify if the issues are ready for development.

> Does your workload for this week look okay?

Yes, the workload for this week feels more comfortable.

> We have two milestones due in the next week. Are there any blockers to meeting our upcoming deadlines?

The preview page #295  is blocked on the PATCH endpoint #562, so I'll start focusing on milestone 5, once I complete all the other issues in m4.

> Anything else you'd like to discuss?

None, at the moment.
